src/Microsoft.TestPlatform.ObjectModel/Framework.cs

```diff
@@ -53,56 +62,71 @@ private Framework()
             return null;
         }
 
-        string name, version;
+        string name, frameworkName, version;
+        string? shortName = null;
         try
         {
             // IDE always sends framework in form of ENUM, which always throws exception
             // This throws up in first chance exception, refer Bug https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_workitems/edit/591142
             var formattedFrameworkString = frameworkString.Trim().ToLower(CultureInfo.InvariantCulture);
+            string? mappedShortName = null;
             switch (formattedFrameworkString)
             {
                 case "framework35":
-                    name = CommonFrameworks.Net35.DotNetFrameworkName;
-                    version = CommonFrameworks.Net35.Version.ToString();
+                    mappedShortName = "net3.5";
                     break;
 
                 case "framework40":
-                    name = CommonFrameworks.Net4.DotNetFrameworkName;
-                    version = CommonFrameworks.Net4.Version.ToString();
+                    mappedShortName = "net4.0";
                     break;
 
                 case "framework45":
-                    name = CommonFrameworks.Net45.DotNetFrameworkName;
-                    version = CommonFrameworks.Net45.Version.ToString();
+                    mappedShortName = "net4.5";
                     break;
 
                 case "frameworkcore10":
-                    name = CommonFrameworks.NetCoreApp10.DotNetFrameworkName;
-                    version = CommonFrameworks.NetCoreApp10.Version.ToString();
+                    mappedShortName = "netcoreapp1.0";
                     break;
 
                 case "frameworkuap10":
-                    name = CommonFrameworks.UAP10.DotNetFrameworkName;
-                    version = CommonFrameworks.UAP10.Version.ToString();
+                    mappedShortName = "uap10.0";
                     break;
+            }
 
-                default:
-                    var nugetFramework = NuGetFramework.Parse(frameworkString);
-                    if (nugetFramework.IsUnsupported)
-                        return null;
+            if (mappedShortName != null)
+            {
+                frameworkString = mappedShortName;
+            }
 
-                    name = nugetFramework.DotNetFrameworkName;
-                    version = nugetFramework.Version.ToString();
+            var nugetFramework = NuGetFramework.Parse(frameworkString);
+            if (nugetFramework.IsUnsupported)
+                return null;
 
-                    break;
+            // e.g. .NETFramework,Version=v3.5
+            name = nugetFramework.DotNetFrameworkName;
+            // e.g. net35
+            try
+            {
+                // .NETPortable4.5 for example, is not a valid framework
+                // and this will throw.
+                shortName = nugetFramework.GetShortFolderName();
```

Comment:
This throws in some cases when the framework is invalid, parsing will return it as Unsupported, but for example if we construct Portable 4.5 and don't specify Profile, it will construct it as **supported**, but throw when we ask for short name. We use short name as folder for logs, which we figure out when we have a "valid" runtime identifier, so keeping this null should be okay, we check for it in all the places where we use it, and supply the original value.